### DISCUSSION TITLE
Filament sensor.

### DISCUSSION THREAD

**Discussion by SteveBz** [1coderookie/KobraGoNeoInsights#2] [Ideas]
I pretty much agree with your assessment of the filament sensor, however, I'm thinking about getting the mod that connects it directly to OctoPrint, then I don't have to worry about connecting to the motherboard.

**Comment by 1coderookie** [comment]
Hi Steve, 
thanks for reaching out and giving me a star also! :)

I didn't add the pictures of the sensor yet, but I do have them already waiting to be uploaded. Connecting it to the mb shouldn't worry you tho as it really is just about plugging it in. Then one doesn't necessarily has to run additional hard- and software like RPi+Octo as well, so that's a bit of an advantage imho.  
However, I don't know about the mod for connecting it to the Pi actually, so if you could give me a link about that, that would be nice. So would that be with the original sensor? It actually sounds like you could use just every sensor for the Pi then, right?  

I currently switched to Klipper an still don't installed that sensor yet tho as I wasn't really able to print using the stock-fw before switching to Klipper, so until now there wasn't really the need to add that sensor for me.

In case you do that mod, a picture and maybe a little description would also be much appreciated if you're willing to do so. In that case just send me an email then please, is easier for me somehow to keep my stuff organized.. ;)

**Reply by SteveBz** [reply]
Wow, you moved to Klipper.  I'm still plucking up courage to do that.

Yes, any sensor for OctoPi.  I'll find a link.

**Reply by 1coderookie** [reply]
The best decision ever tbh. First time I'm getting beautiful repeatable 1st layers and being able to finally *print*. No restrictions of the stock-fw and not to mention being able to use Pressure Advance and (what I didn't do yet, need to solder that) Input Shaping.. 
And you can ref-flash the stock fw any time btw..

**Comment by SteveBz** [comment]
Here is the OctoPi / filament sensor starter.  You have to drill down from there:

https://all3dp.com/2/octoprint-filament-sensor-simply-explained/

Can you point me at the Klipper instructions you followed, please?

**Reply by 1coderookie** [reply]
Great, thanks. 

Sure, it's the reddit-link in the Klipper chapter of the pages: https://www.reddit.com/r/anycubic/comments/10cwm16/install_klipper_on_kobra_go_or_neo/
It recently was added to the main Klipper repo, so no need to use the Steve Gotthart repo anymore. But that's also mentioned there.

Btw: How did you find my page? Google doesn't really find it yet (guess I need to get my hands down on a bit of SEO..).

**Comment by SteveBz** [comment]
I don't recall, but Google I think, but probably not on the first page. I was looking for Neo mods. 

⁣Get BlueMail for Android ​

**Reply by 1coderookie** [reply]
Oh really? Cool! Ok, so I need to dial into SEO then haha

**Comment by SteveBz** [comment]
I promised I'd tell you when I added the filament sensor.

Need to finish testing and run the cabling properly.  I've attached it to pins 17, 18 and 20 on the GPIO and the OctoPi concludes successfully.  I just haven't tested a full runout during a print.

![JPEG_20230320_081208_](https://user-images.githubusercontent.com/1646078/226382242-18ee4321-35bd-4e41-8cef-61af5a27fb7f.jpg)

But the initial test worked!

Kind regards

Steve

**Reply by 1coderookie** [reply]
Cool solution if one has a sensor laying around already which doesn't fit the mainboard!

**Reply by SteveBz** [reply]
I'm not sure that it would work at all if it did fit the main board.  Klipper or OctoPrint would have to work out what had happened and deal with it.  It seems much cleaner to plug straight into the Pi.  I think plugging into the printer board is for standalone mode.

**Reply by 1coderookie** [reply]
Of course it would also work out with Klipper if you configure it correctly, same way as with the stock firmware..

**Comment by SteveBz** [comment]
So amazingly I just had a filament break and it paused, allowed me to fix it and resume.

**Comment by ntguest** [comment]
you can use any cheapest sensors (i bought one for less than dollar on ali)
... and you can can share my [sensors mount](https://www.printables.com/model/505135) for go

**Reply by 1coderookie** [reply]
Is the one you bought for $1 the one at the picture? Cuz that one looks exactly like the one AC sells!  
Thanks, I didn't look out yet for models to mount it differently than it's supposed to, so I'll add it asap.

**Reply by ntguest** [reply]
here is https://www.aliexpress.com/item/1005003397498506.html
![image](https://github.com/1coderookie/KobraGoNeoInsights/assets/69485846/732e43d3-78ab-444d-95ea-1bfa6aed2e09)
use only two pins - GND and SIG

**Reply by 1coderookie** [reply]
LOL - it *IS* the one AC sells for 20 bucks! Thanks!

**Reply by ntguest** [reply]
but they sell the cheapest printers)

**Comment by SteveBz** [comment]
Yes it's the one I use too. It cost about 3 pounds on ebay.

⁣Get BlueMail for Android ​

**Comment by NuclearPhoenixx** [comment]
Does the printer pause the print if the sensor detects no filament or does it just halt it without the possibility of resuming? Because if that's the case then it's completely useless IMO. Except maybe in cases where you print huge things I wouldn't care if the printer just goes on "dry" without filament instead of buying that $20 thing from Anycubic (I know you can buy the same thing on AliExpress for like $1). Or does that harm the hotend in any way?

I mean you'd only need it once per spool max anyways, depending on how much you print of course that can be a long time.

**Reply by 1coderookie** [reply]
Well, tbh: I don't even know exactly as I never tried it myself.  
I mean, it *should* pause the print and actually it *should* shut off the cartridge heater while keeping the bed warm, but I highly doubt it really does so when using the stock fw. I *think* it only throws a warning, maybe it even stops - but in that case it most likely will shut off the bed as well. At least this wouldn't surprise me, looking at certain bugs and wrong settings anyway.. *cough cuogh*  

I actually never used the sensors, even tho I have them. If I'm in doubt if there's enough filament on the spool, I just put it on a kitchen scale and do the rough calculation if I'm most likely on the safe side or not. If it's really close, I'll be close to the printer and have an eye on it, so that I can feed a filament from a new spool right in. My 'best' and closest print was with having ~20cm left on the spool after it finished, so I made that a bit of a challenge for myself haha 

Problem with the sensor is that they often cause false positive detections, so they aren't working reliable anyway.  
And if you have filament on the spool tangled up and it just blocks, it wouldn't detect that anyway due to the design itself - cuz in that case there would still be filament loaded. Means, it would just continue printing in the air without achieving anything else than cooking the filament which is in there.. 

It's (said to be) not good to leave the hotend heated up too long without filament being loaded in there, and it's also not good doing so with having filament loaded but not printing, as that'll just cook and burn the filament then.

**Reply by NuclearPhoenixx** [reply]
Ok, thanks, good to know. That really sounds not too helpful and that's what I thought considering the false positives too, ugh. It's really not too hard to guess the remaining filament on the spool so yeah...

**Reply by 1coderookie** [reply]
Yeah, imho don't waste your money on that - better buy a RPi Zero2W for those 20$ and use Klipper instead, that money will be invested better ;)

**Comment by SteveBz** [comment]
It's true, that you almost never use it. I've only used mine once, but it does pause and allow resume.

⁣Get BlueMail for Android ​

**Reply by 1coderookie** [reply]
Thx - and does it actually shut off the cartridge heater while keeping the bed hot? Or is it shutting off everything? Or keeps everything hot?

**Comment by SteveBz** [comment]
Sorry, I don't recall.  

Steve.
